integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java

```diff
@@ -762,85 +729,161 @@ public void testMiiChangeSSL() {
-            MODEL_DIR + "/model.jdbc2.yaml", pathToRemoveTargetYaml.toString()), withStandardRetryPolicy);
+            MODEL_DIR + "/model.jdbc2.updatejdbcdriverparams.yaml", pathToChangReadsYaml.toString()),
+              withStandardRetryPolicy);
 
-    // Patch a running domain with introspectVersion.
+    // Patch a running domain with onNonDynamicChanges - update with CommitUpdateOnly so that even if previous test
+    // updates onNonDynamicChanges, this test will work
+    patchDomainResourceWithOnNonDynamicChanges(domainUid, domainNamespace, "CommitUpdateOnly");
+
+    // Patch a running domain with introspectVersion, uses default value for onNonDynamicChanges
     String introspectVersion = patchDomainResourceWithNewIntrospectVersion(domainUid, domainNamespace);
 
     // Verifying introspector pod is created, runs and deleted
     verifyIntrospectorRuns();
 
+    // Verify domain is not restarted when non-dynamic change is made using default CommitUpdateOnly
+    verifyPodsNotRolled(pods);
+
     verifyPodIntrospectVersionUpdated(pods.keySet(), introspectVersion);
 
-    // make sure the application is not deployed on cluster
-    verifyApplicationRuntimeOnCluster("404");
+    // check the change is committed
+    int adminServiceNodePort
+        = getServiceNodePort(domainNamespace, getExternalServicePodName(adminServerPodName), "default");
+    assertNotEquals(-1, adminServiceNodePort, "admin server default node port is not valid");
 
-    // make sure the application is not deployed on admin
-    withStandardRetryPolicy.conditionEvaluationListener(
-        condition ->
-            logger.info("Waiting for application target to be updated. "
-                    + "Elapsed time {0}ms, remaining time {1}ms",
-                condition.getElapsedTimeInMS(), condition.getRemainingTimeInMS())).until(
-                  () -> checkApplicationRuntime(domainNamespace, adminServerPodName,
-            adminServerName, "404"));
+    // check server config for ScatteredReadsEnabled is updated
+    assertTrue(checkSystemResourceConfig(adminServiceNodePort,
+        "servers/" + adminServerName,
+        "\"scatteredReadsEnabled\": true"), "ScatteredReadsEnabled is not changed to true");
+    logger.info("ScatteredReadsEnabled is changed to true");
+
+    // check datasource configuration using REST api
+    assertTrue(checkSystemResourceConfig(adminServiceNodePort,
+        "JDBCSystemResources/TestDataSource2/JDBCResource/JDBCDriverParams/properties/properties",
+        "\"name\": \"testattrib\""), "JDBCSystemResource new property not found");
+    logger.info("JDBCSystemResource new property found");
+
+    // check that the domain status condition type is "ConfigChangesPendingRestart"
+    // and message contains the expected msg
+    logger.info("Verifying the domain status condition message contains the expected msg");
+    verifyDomainStatusCondition("ConfigChangesPendingRestart", expectedMsgForCommitUpdateOnly);
```

Comment:
Can we add a test to verify each pod has the label 'weblogic.configurationChangedPendingRestart=true' set and the same label should be gone after the domain is restarted?

Reply:
LabelConstants.MII_UPDATED_RESTART_REQUIRED_LABEL
 
   String MII_UPDATED_RESTART_REQUIRED_LABEL = "weblogic.configurationChangedPendingRestart";

Reply:
I can add the lable check to this test